ui/app/components/app/signature-request/signature-request.component.js

```diff
@@ -0,0 +1,88 @@
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+import Header from './signature-request-header'
+import Footer from './signature-request-footer'
+import Message from './signature-request-message'
+import { ENVIRONMENT_TYPE_NOTIFICATION } from './signature-request.constants'
+import { getEnvironmentType } from '../../../../../app/scripts/lib/util'
+import Identicon from '../../ui/identicon'
+
+export default class SignatureRequest extends PureComponent {
+  static propTypes = {
+    txData: PropTypes.object.isRequired,
+    accounts: PropTypes.shape(
+      PropTypes.shape({
+        address: PropTypes.string,
+        balance: PropTypes.string,
+      })
+    ),
+    selectedAccount: PropTypes.shape({
+      address: PropTypes.string,
+      balance: PropTypes.string,
+      name: PropTypes.string,
+    }).isRequired,
+
+    clearConfirmTransaction: PropTypes.func.isRequired,
+    cancel: PropTypes.func.isRequired,
+    sign: PropTypes.func.isRequired,
+  }
+
+  static contextTypes = {
+    t: PropTypes.func,
+  }
+
+  componentDidMount () {
+    const { clearConfirmTransaction, cancel } = this.props
+    const { metricsEvent } = this.context
+    if (getEnvironmentType(window.location.href) === ENVIRONMENT_TYPE_NOTIFICATION) {
+      window.onbeforeunload = event => {
```

Comment:
As @Gudahtt has pointed out elsewhere (https://github.com/MetaMask/metamask-extension/pull/7333 and https://github.com/MetaMask/metamask-extension/pull/7335):

> Generally it's better to use `window.addEventListener('beforeunload', (event) => { ... });` instead of `window.onbeforeunload` because it avoids overwriting other `onbeforeunload` handlers